execute/plugin.go

```diff
@@ -295,6 +297,32 @@ func (p *Plugin) Reports(
 func (p *Plugin) ShouldAcceptAttestedReport(
 	ctx context.Context, u uint64, r ocr3types.ReportWithInfo[[]byte],
 ) (bool, error) {
+	// check if we support the dest, if not we can't do the checks needed.
+	supports, err := p.chainSupport.SupportsDestChain(p.reportingCfg.OracleID)
+	if err != nil {
+		return false, fmt.Errorf("supports dest chain: %w", err)
+	}
+
+	if !supports {
+		p.lggr.Warnw("dest chain not supported, can't run report acceptance procedures")
+
+		// TODO: return false here or a non-nil error?
+		return false, nil
+	}
+
+	offRampConfigDigest, err := p.ccipReader.GetOffRampConfigDigest(ctx, consts.PluginTypeExecute)
+	if err != nil {
+		return false, fmt.Errorf("get offramp config digest: %w", err)
+	}
+
+	if !bytes.Equal(offRampConfigDigest[:], p.reportingCfg.ConfigDigest[:]) {
+		p.lggr.Warnw("my config digest doesn't match offramp's config digest, not accepting report",
+			"myConfigDigest", p.reportingCfg.ConfigDigest,
+			"offRampConfigDigest", hex.EncodeToString(offRampConfigDigest[:]),
+		)
+		return false, nil
+	}
+
 	// Just a safety check, should never happen.
 	if r.Report == nil {
```

Comment:
This should be the first check, no? We should optimize to do "pure" checks first (ones that don't require IO operations)

Similar to this one
```go
	if len(decodedReport.ChainReports) == 0 {
		p.lggr.Info("skipping empty report")
		return false, nil
	}
```

Reply:
Good find, will rejig stuff a bit

Reply:
Fixed :)